nix/nixos/smash-service.nix

```diff
@@ -20,39 +29,51 @@ in {
         type = lib.types.attrs;
         default = cfg.environment.explorerConfig;
       };
-      environment = lib.mkOption {
+      nodeConfig = lib.mkOption {
         type = lib.types.attrs;
-        default = pkgs.iohkNix.cardanoLib.environments.${cfg.environmentName};
+        default = cfg.environment.nodeConfig;
+      };
+      environment = lib.mkOption {
+        type = lib.types.nullOr lib.types.attrs;
+        default = self.iohkNix.cardanoLib.environments.${cfg.environmentName};
       };
       logConfig = lib.mkOption {
         type = lib.types.attrs;
-        default = pkgs.iohkNix.cardanoLib.defaultExplorerLogConfig;
+        default = self.iohkNix.cardanoLib.defaultExplorerLogConfig;
       };
       environmentName = lib.mkOption {
         type = lib.types.str;
         description = "environment name";
       };
       socketPath = lib.mkOption {
-        type = lib.types.path;
+        type = lib.types.nullOr lib.types.path;
+        default = null;
       };
       user = lib.mkOption {
         type = lib.types.str;
         default = "smash";
         description = "the user to run as";
       };
       postgres = {
+        generatePGPASS = lib.mkOption {
+          type = lib.types.bool;
+          default = true;
+          description = "generate pgpass";
+        };
+
         pgpass = lib.mkOption {
           type = lib.types.path;
           default = builtins.toFile "pgpass" "${cfg.postgres.socketdir}:${toString cfg.postgres.port}:${cfg.postgres.database}:${cfg.postgres.user}:*";
         };
+
         socketdir = lib.mkOption {
           type = lib.types.str;
           default = "/run/postgresql";
           description = "the path to the postgresql socket";
         };
         port = lib.mkOption {
           type = lib.types.int;
-          default = config.services.postgresql.port;
```

Comment:
I'm not sure how to make this work with `lib.evalModules` so I changed default to 5432.